src/platform/i386/chal_pgtbl.c

```diff
@@ -660,7 +661,11 @@ chal_pgtbl_pgtblactivate(struct captbl *ct, capid_t cap, capid_t pt_entry, capid
 	ret = cap_kmem_activate(ct, pgtbl_cap, kmem_cap, (unsigned long *)&kmem_addr, &pte);
 	if (unlikely(ret)) return ret;
 	assert(kmem_addr && pte);
- 
+#if defined(__i386__)	
+	/* x86-32 does not support PCID :( */
+	if (asid != 0) return -EINVAL;
```

Comment:
maybe it is better to just proceed. x86-32 can pass in any asids it wants. The system will still be functionally correct. So it might be better to simply have this code for 32 be the same. In the kernel, we just ignore the value.

We don't want to have to maintain differnet client code for 32 bit vs. 64.